src/snes/supermetroid/modules/RidleyOverkillTracker.js

```diff
@@ -0,0 +1,139 @@
+import MemoryModule from "../../../util/memory/MemoryModule";
+import { Rooms } from "../enums";
+import Addresses from "../addresses";
+
+export default class RidleyOverkillTracker extends MemoryModule {
+    constructor() {
+        super("ridleyOverkillTracker", "Ridley Overkill Tracker");
+        this.tooltip = "Writes in Chat how many shots you wasted on Ridley.";
+        this.reset();
+    }
+
+    settingDefs = {
+        displayDoorTimes: {
+            display: "Use a custom event (name: ridleyOverkill) instead of default message",
+            type: "bool",
+            default: false,
+        }
+    };
+
+    setEnabled(enabled) {
+        super.setEnabled(enabled);
+        this.reset();
+    }
+
+    shouldRunForGame(gameTags) {
+        return gameTags.SM;
+    }
+
+    getMemoryReads() {
+        return [
+            Addresses.roomID,
+            Addresses.gameState,
+            Addresses.enemyHP,
+            Addresses.samusHP,
+            Addresses.samusReserveHP,
+            Addresses.ridleyIframeTimer,
+            Addresses.ridleyShotDamage,
+        ];
+    }
+
+    reset() {
+        this.lastCheck = 0;
+        this.inRidleyFight = false;
+        this.startCounting = false;
+        this.additionalShotCounter = 0;
+        this.additionalDamageDealt = 0;
+    }
+
+    getMessage() {
+        return 'That was an overkill. ' +
+            'You had ' + this.additionalShotCounter + ' ' +
+            'additional <{plural("shot", ' + this.additionalShotCounter + ')}>'
+    }
+
+    memoryReadAvailable({ memory, sendEvent }) {
+        // Handle a run being reset
+        if (
+            memory.roomID.prevFrameValue !== undefined &&
+            memory.roomID.prevFrameValue !== Rooms.EMPTY &&
+            memory.roomID.value === Rooms.EMPTY
+        ) {
+            if (this.inRidleyFight) {
+                // were in ridley fight but reset due to ridley
+            }
+            this.reset();
+            return;
+        }
```

Comment:
So the ResetEvent module is slated to run before this one, and it sets a `globalState.isReset = true` value for this current memory read cycle. You can use that in other modules by including `globalState` as one of the destructured params for the function. You can then just check `if (globalState.isReset) { ...` to check for resets properly without duplicating any logic.

```suggestion
    memoryReadAvailable({ memory, sendEvent, globalState }) {
        // Handle a run being reset
        if (globalState.isReset) {
            this.reset();
            return;
        }
```